src/decompiler/__snapshots__/decompileAll.spec.ts.snap

```diff
@@ -1835,3 +1835,129 @@ exports[`decompileAll should decompile wallet v4 1`] = `
   }>
 }END>c"
 `;
+
+exports[`decompileAll should decompile mathlib.fc 1`] = `
```

Comment:
It does not look like this covers the whole `mathlib.fc`: https://github.com/ton-blockchain/ton/blob/5c392e0f2d946877bb79a09ed35068f7b0bd333a/crypto/smartcont/mathlib.fc, which is almost a thousand lines of code

Reply:
that's right, it doesn't. I guess it might be enough in context of the original issue, but I can make it to include all of the functions by calling them in `main()`.

Reply:
yeah, let's include the whole thing into the test

Reply:
Just tried compiling the whole thing with func optimizations turned off so that all of the functions are included into result. Test failed but not because of `ton-opcode` but becaues of `@ton/core` boc deserialization:
```
Invalid magic

      222 |     let src: Cell;
      223 |     if (Buffer.isBuffer(args.src)) {
    > 224 |         src = Cell.fromBoc(args.src)[0];
          |                    ^
      225 |     } else {
      226 |         src = args.src;
      227 |     }

      at parseBoc (node_modules/@ton/core/dist/boc/cell/serialization.js:160:15)
      at deserializeBoc (node_modules/@ton/core/dist/boc/cell/serialization.js:168:15)
      at Function.fromBoc (node_modules/@ton/core/dist/boc/Cell.js:34:51)
      at decompileAll (src/decompiler/decompileAll.ts:224:20)
      at Object.<anonymous> (src/decompiler/decompileAll.spec.ts:70:31)
```

Reply:
oh, wow! let's open an issue in the `@ton/core` repo then and link it here

Reply:
oh, my bad! I forgot that `func` writes a fift output by default, not BOC. I'll replace the file.